DelegateUIKit/ProductSelectionVC.swift

```diff
@@ -0,0 +1,140 @@
+//
+//  ProductSelectionVC.swift
+//  DelegateUIKit
+//
+//  Created by Erwin Warkentin on 08.03.23.
+//
+
+import UIKit
+
+//This Protocol tells, the Controller who subscribed to this protocal, what is to do
+protocol ProductSelectionDelegate {
+    func didSelectProduct(name: String, imageName: String)
+}
+
+class ProductSelectionVC: UIViewController {
+    
+    let iPhoneButton = UIButton()
+    let iPadButton = UIButton()
+    let appleWatchButton = UIButton()
+    let macBookButton = UIButton()
+    let clearButton = UIButton()
+    
+    //tells the viewController what the delegate is
+    var delegate: ProductSelectionDelegate?
```

Comment:
Like mentioned before, this should be a weak reference `weak var delegate: ProductSelectionDelegate?` to prevent a retain cycle.